metagpt/actions/action.py

```diff
@@ -16,13 +16,16 @@
 from metagpt.logs import logger
 from metagpt.utils.common import OutputParser
 from metagpt.utils.custom_decoder import CustomDecoder
-
+from metagpt.provider.openai_api import OpenAIGPTAPI
+from metagpt.provider.customized_api import CustomizedGPTAPI
+from metagpt.config import CONFIG
+import openai
 
 class Action(ABC):
     def __init__(self, name: str = "", context=None, llm: LLM = None):
         self.name: str = name
         if llm is None:
-            llm = LLM()
+            llm = LLM(self.__str__())    # Pass in Action name for LLM selection.
```

Comment:
self.__str__() is a string like `<__main__.XXAction object at 0x7fe165a2a7c0>`, not the ClassName.

Reply:
In fact self.__str__() will return self.__class__.__name__ in Action class function

Reply:
> In fact self.**str**() will return self.**class**.**name** in Action class function

>>> class A(ABC):
...     def __init__(self):
...         print(type(self.__str__()))
...         print(self.__str__())
...
>>> a= A()
<class 'str'>
<__main__.A object at 0x7fed58072ec0>

Reply:
There are the following methods in Action
def __str__(self).
        return self.__class__. __name__
actually returns the name of the Action, not <main.A object at 0x7fed58072ec0>. 

A = WritePRD()
print(A.__str__())
WritePRD